### PR TITLE
feat(dashmate): setup masternode with DMT

### PR DESCRIPTION
**Pull request by shumkov**
## Issue being fixed or feature implemented

Dash Masternode Tool is recommended way to register masternode for mainnet but it was missed in dashmate setup

## What was done?

- Add DMT as a choice to register a masternode
- Disabled mobile wallets since they do not provide all required functionality
- Fixed initial values for IP and ports form
- Disable e2e tests (not stable)

## How Has This Been Tested?

Manually

## Breaking Changes

None

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Review by jawid-h — DISMISSED** [review]

File: packages/dashmate/src/listr/prompts/createIpAndPortsForm.js
Comment (thephez) [inline packages/dashmate/src/listr/prompts/createIpAndPortsForm.js:1]: When I ran through the DMT (testnet) setup, it didn't pre-fill the ports with default values like it used to. I'm not sure if that was intentional or not, but I liked having some defaults there.

File: packages/dashmate/src/listr/prompts/registerMasternode/registerMasternodeWithDMT.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      separator: () => '',
+      format: () => '',
+      initial: true,
+      isTrue: () => true,
+    },
+    {
+      type: 'input',
+      name: 'operatorPrivateKey',
+      header: `  Dashmate needs to collect details on the operator key. The operator
+  key is a BLS private key, encoded in HEX format. Dashmate will record the
```
Comment (thephez) [inline packages/dashmate/src/listr/prompts/registerMasternode/registerMasternodeWithDMT.js:47]:
```suggestion
  key is a BLS private key, encoded in hexadecimal format. Dashmate will record the
```

File: packages/dashmate/src/listr/tasks/setup/regular/registerMasternodeGuideTaskFactory.js [hunk 2]
```diff
@@ -35,15 +14,13 @@ function registerMasternodeGuideTaskFactory() {
...
-    const blsSignatures = await BlsSignatures();
-
-    const validateBLSPrivateKey = validateBLSPrivateKeyFactory(blsSignatures);
-
     const REGISTRARS = {
       CORE: 'dashCore',
-      ANDROID: 'dashAndroid',
-      IOS: 'dashIOS',
-      OTHER: 'other',
+      // TODO: Disabled until additional functionality like signing protx and so on is not
```
Comment (thephez) [inline packages/dashmate/src/listr/tasks/setup/regular/registerMasternodeGuideTaskFactory.js:19]:
```suggestion
      // TODO: Disabled until additional functionality like signing protx and so on is
```

**Review by thephez — COMMENTED** [review]
I tested this a bit and left a few suggestions.

Also I noticed a couple small unexpected things:

- If you've already registered your MN, it will require you to enter the BLS key as expected with no pre-generated key in the field. In the next step, it then asks for the platform key, but actually shows a generated key. This should be unnecessary for already registered nodes (because they've already defined a key when registering).
- When I ran through the DMT (testnet) setup, it didn't pre-fill the ports with default values like it used to. I'm not sure if that was intentional or not, but I liked having some defaults there.

**Comment by shumkov** [comment]
@thephez Thank you for your comments and suggestions. I will make fixes tomorrow.

**Comment by shumkov** [comment]
> When I ran through the DMT (testnet) setup, it didn't pre-fill the ports with default values like it used to. I'm not sure if that was intentional or not, but I liked having some defaults there.

You are supposed to use once what you set in DMT, so that's why they are empty.

File: packages/dashmate/src/listr/prompts/registerMasternode/registerMasternodeWithCoreWallet.js [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+    const { privateKey: initialOperatorPrivateKey } = await generateBlsKeys();
+
+    const prompts = [
+      {
+        type: 'form',
+        name: 'collateral',
+        header: `  Dashmate needs to collect your collateral funding transaction hash and index.              
+  The funding value must be exactly ${collateralAmount} ${collateralDenomination}.
+  
+  Please follow instructions on how to send a collateral in Dash Core Wallet:
```
Comment (strophy) [inline packages/dashmate/src/listr/prompts/registerMasternode/registerMasternodeWithCoreWallet.js:87]:
```suggestion
  Please follow the instructions on how to create a collateral funding transaction in Dash Core Wallet:
```

File: packages/dashmate/src/listr/prompts/registerMasternode/registerMasternodeWithDMT.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+ * }>}
+ */
+async function registerMasternodeWithDMT(ctx, task) {
+  const blsSignatures = await BlsSignatures();
+  const validateBLSPrivateKey = validateBLSPrivateKeyFactory(blsSignatures);
+
+  const prompts = [
+    {
+      type: 'confirm',
+      header: `  Complete the initial DMT setup then return here to continue:
```
Comment (strophy) [inline packages/dashmate/src/listr/prompts/registerMasternode/registerMasternodeWithDMT.js:32]:
```suggestion
      header: `  Complete initial DMT setup and return here to continue:
```

**Review by strophy — CHANGES_REQUESTED** [review]

**Review by strophy — APPROVED** [review]

**Review by strophy — DISMISSED** [review]

**Review by strophy — APPROVED** [review]

**Review by strophy — APPROVED** [review]

**Pull request merged**